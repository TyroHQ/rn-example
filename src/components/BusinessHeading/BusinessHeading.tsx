import React from "react";
import { Text, View } from "react-native";
import styled from 'styled-components'
import { Placeholder } from "../Placeholder";

const BusinessHeadingContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`

export class BusinessHeading extends React.Component<{}> {
  static defaultProps: BusinessHeading["props"];
  render() {
    return <BusinessHeadingContainer>
      <Placeholder>Business Logo</Placeholder>
      <Placeholder>Cover Business Info</Placeholder>
      <Placeholder>Cover Photo</Placeholder>
    </BusinessHeadingContainer>
  }
}
