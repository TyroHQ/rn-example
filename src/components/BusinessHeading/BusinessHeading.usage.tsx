import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { BusinessHeading } from "./index";

export const basic = () => <BusinessHeading />;
