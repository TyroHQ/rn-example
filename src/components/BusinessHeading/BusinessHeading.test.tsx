import { shallow } from "enzyme";
import * as usage from "./BusinessHeading.usage";

describe("<BusinessHeading />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
