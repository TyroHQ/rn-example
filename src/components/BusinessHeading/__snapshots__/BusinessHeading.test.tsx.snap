// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessHeading /> renders well in basic mode 1`] = `
<Styled(Component)>
  <Placeholder
    orientation="row"
  >
    Business Logo
  </Placeholder>
  <Placeholder
    orientation="row"
  >
    Cover Business Info
  </Placeholder>
  <Placeholder
    orientation="row"
  >
    Cover Photo
  </Placeholder>
</Styled(Component)>
`;
